### PR TITLE
Group Prices

### PR DESCRIPTION
**Pull request by aligentjim**

### CODE REVIEW

File: app/code/community/Aligent/CacheObserver/Model/Observer.php [hunk 1]
```diff
@@ -38,9 +35,11 @@ public function customBlockCache(Varien_Event_Observer $observer) {
                     return $this;
                 }
             }
+            $iCacheLifetime = Mage::getStoreConfig(self::CACHE_LIFETIME_CONFIG);
```
Comment (aligentjim) [inline app/code/community/Aligent/CacheObserver/Model/Observer.php]: @moloughlin This observer method gets called many times in the course of rendering a page, and most of the blocks we observe aren't made cacheable by cacheobserver.  However we're doing this lookup into the XML tree for the cache lifetime on every block regardless of whether we subsequently act on the block.  The observer class is instantiated as a singleton.  Can we make $iCacheLifetime a class level member variable and set it in the constructor instead?  That way we only make one XML lookup for the entire page load.

**Comment by aligentjim** [comment]
@moloughlin Can we also add a system config value that enables/disabled the group pricing functionality.  When disabled we should still retain the existing "logged in/logged out" status in the cache key.  This should be disabled by default so there's no change in functionality for existing sites that are upgrading.

**Comment by moloughlin** [comment]
@aligentjim I've made those updates and it should be ready for merge now if you want to give it another quick look. Cheers

File: app/code/community/Aligent/CacheObserver/Model/Observer.php [hunk 2]
```diff
@@ -104,17 +113,16 @@ public function customBlockCache(Varien_Event_Observer $observer) {
                 }
                 $vPageParamKey = $this->getParamKey(self::PAGE_VAR);
                 $vAlias = $block->getNameInLayout();
-                $block->setData('cache_lifetime', self::CUSTOM_CACHE_LIFETIME);
-                $block->setData('cache_key', 'catalog_product_abstractview_product_' . $iProductId.'_'.$vPageParamKey.(Mage::getSingleton('customer/session')->isLoggedIn() ? '_loggedin' : '_loggedout') . '_store_' . Mage::app()->getStore()->getId() . '_' . Mage::app()->getStore()->getCurrentCurrencyCode().'_'.$vAlias);
-                $block->setData('cache_tags', array(Mage_Core_Block_Abstract::CACHE_GROUP, Mage_Core_Model_App::CACHE_TAG, Mage_Core_Model_Store::CACHE_TAG, Mage_Catalog_Model_Product::CACHE_TAG.'_'.$iProductId));
+                $block->setData('cache_lifetime', $this->iCacheLifetime);
+                $block->setData('cache_key', 'catalog_product_abstractview_product_' . $iProductId.'_'.$vPageParamKey.(Mage::getSingleton('customer/session')->isLoggedIn() ? Mage::getSingleton('customer/session')->getCustomerGroupId() : '0') . '_store_' . Mage::app()->getStore()->getId() . '_' . Mage::app()->getStore()->getCurrentCurrencyCode().'_'.$vAlias);
```
Comment (lukenm) [inline app/code/community/Aligent/CacheObserver/Model/Observer.php]: @aligentjim Is it possible to have different groups for not logged in customers?

Comment (aligentjim) [inline app/code/community/Aligent/CacheObserver/Model/Observer.php]: @lukenm I'm not 100% sure, but I don't think so.

**Pull request merged**